samples/nrf9160/modem_shell/src/link/link_shell.c

```diff
@@ -1741,6 +1743,1441 @@ int link_shell(const struct shell *shell, size_t argc, char **argv)
+					"RPTAU bit string length must be %d.",
+					LINK_SHELL_PSM_PARAM_STR_LENGTH);
+				return -EINVAL;
+			}
+			break;
+		case 't': /* rat */
+			if (strlen(optarg) ==
+			    LINK_SHELL_PSM_PARAM_STR_LENGTH) {
+				strcpy(psm_rat_bit_str, optarg);
+				psm_rat_set = true;
+			} else {
+				mosh_error(
+					"RAT bit string length must be %d.",
+					LINK_SHELL_PSM_PARAM_STR_LENGTH);
+				return -EINVAL;
+			}
+			break;
+		default:
+			break;
+		}
+	}
+
+	if (common_option == LINK_COMMON_ENABLE) {
+		/* Set with the defaults if not given */
+		char *rptau_bit_value = NULL;
+		char *rat_bit_value = NULL;
+
+		if (psm_rptau_set) {
+			rptau_bit_value = psm_rptau_bit_str;
+		}
+
+		if (psm_rat_set) {
+			rat_bit_value = psm_rat_bit_str;
+		}
+
+		ret = lte_lc_psm_param_set(rptau_bit_value,
+						rat_bit_value);
+		if (ret < 0) {
+			mosh_error("Cannot set PSM parameters: error %d", ret);
+			mosh_error(
+				"  rptau %s, rat %s",
+				((rptau_bit_value == NULL) ? "NULL" : rptau_bit_value),
+				((rat_bit_value == NULL) ? "NULL" : rat_bit_value));
+			return -EINVAL;
+		}
+
+		ret = lte_lc_psm_req(true);
+		if (ret < 0) {
+			mosh_error("Cannot enable PSM: %d", ret);
+		} else {
+			mosh_print("PSM enabled");
+		}
+	} else if (common_option == LINK_COMMON_DISABLE) {
+		ret = lte_lc_psm_req(false);
+		if (ret < 0) {
+			mosh_error("Cannot disable PSM: %d", ret);
+		} else {
+			mosh_print("PSM disabled");
+		}
+	} else if (common_option == LINK_COMMON_READ) {
+		int tau, active_time;
+
+		ret = lte_lc_psm_get(&tau, &active_time);
+		if (ret < 0) {
+			mosh_error("Cannot get PSM configs: %d", ret);
+		} else {
+			mosh_print(
+				"PSM config: TAU %d %s, active time %d %s",
+				tau,
+				(tau == -1) ? "(timer deactivated)" :
+				"seconds",
+				active_time,
+				(active_time == -1) ?
+				"(timer deactivated)" :
+				"seconds");
+		}
+	} else {
+		link_shell_print_usage(LINK_CMD_PSM);
+	}
+
+	return 0;
+}
+
+static int link_shell_rai(const struct shell *shell, size_t argc, char **argv)
+{
+	enum link_shell_common_options common_option = LINK_COMMON_NONE;
+
+	common_option = link_shell_getopt_common(argc, argv);
+
+	if (common_option == LINK_COMMON_READ) {
+		link_rai_read();
+	} else if (common_option == LINK_COMMON_ENABLE) {
+		link_rai_enable(true);
+	} else if (common_option == LINK_COMMON_DISABLE) {
+		link_rai_enable(false);
+	} else {
+		link_shell_print_usage(LINK_CMD_RAI);
+	}
+
+	return 0;
+}
+
+static int link_shell_rsrp(const struct shell *shell, size_t argc, char **argv)
+{
+	enum link_shell_common_options common_option = LINK_COMMON_NONE;
+
+	common_option = link_shell_getopt_common(argc, argv);
+
+	if (common_option == LINK_COMMON_SUBSCRIBE) {
+		link_rsrp_subscribe(true);
+	} else if (common_option == LINK_COMMON_UNSUBSCRIBE) {
+		link_rsrp_subscribe(false);
+	} else {
+		link_shell_print_usage(LINK_CMD_RSRP);
+	}
+
+	return 0;
+}
+
+static int link_shell_search(const struct shell *shell, size_t argc, char **argv)
```

Comment:
The `link search --write` command just prints the help, although I think it should write the default configuration to the modem:
```
mosh:~$ link search --write
Usage: link search --read | --write | --clear | --start

Options:
  -r, --read,   Read modem search configuration
      --clear,  Clear modem search configuration
      --write,  Write modem search configuration.
...
```
`link old search --write` works as expected.

Reply:
Good catch. `--write` option handling was missing from `link_shell_getopt_common_option` as it's not used by anything other than this new search command.